### ISSUE TITLE
Node limit reset when adding a callback

### ISSUE THREAD

**Issue by jdeboeck** [JuliaOpt/JuMP.jl#1171]
Hello,

When adding a cutcallbak to a JuMP model and limiting the nodes in the branching tree, the node limit seems to be reset as soon as the solving procedure starts. In the following example : 
```julia
using JuMP
using CPLEX
m = Model(solver=CplexSolver())
@variable(m, 0 <= x[1:2] <= 3.5, Int)
@objective(m, Max, sum{x[i], i in 1:2})
println(m)

function fct(cb)
  cpxNodes = CPLEX.get_param(m.internalModel.inner.env,"CPX_PARAM_NODELIM")
  println("User cut $cpxNodes")                                   ---------------> "9223372036800000000"
end

JuMP.build(m)
addcutcallback(m,fct)
CPLEX.set_param!(m.internalModel.inner.env,"CPX_PARAM_NODELIM",10)
cpxNodes = CPLEX.get_param(m.internalModel.inner.env,"CPX_PARAM_NODELIM")
println("Before solve $cpxNodes")                              ---------------->"10"
cpxNodes = CPLEX.get_param(m.internalModel.inner.env,"CPX_PARAM_NODELIM")
println("Start $cpxNodes")                                            ---------------->"10"
solve(m)
println(getobjectivevalue(m))
cpxNodes = CPLEX.get_param(m.internalModel.inner.env,"CPX_PARAM_NODELIM")
println("End $cpxNodes")                                              ---------------->"9223372036800000000"
```
I obtain the outputs in quotes. The quote in "fct" is obtained on bigger instances. If I delete the sign adding the call back, the CPX_PARAM_NODELIM stays at 10, when cuts are added and after the solving. As I never had this issue programming in C++ not being able to add userCuts and limit the number of nodes, I guess there as an issue when adding the cut that forces to use default parameter values ?

**Comment by jdeboeck** [comment]
There is the possibility to go around this issue by using the return JuMP.StopTheSolver from branch release-0.18 (discussed in #1144) at the end of the callBack but I guess this is not the most elegant way !

**Comment by mlubin** [comment]
Does this still happen if you set the parameter as `CplexSolver(CPX_PARAM_NODELIM=10)`?

The callback model is going to change entirely from JuMP 0.19 and forward, so it's not a high priority to address issues with the old system. We'll accept fixes on the `release-0.18` branch if you figure out what's happening.

**Comment by jdeboeck** [comment]
Thank you. With CplexSolver(CPX_PARAM_NODELIM=10) the limit is well respected. Next to observing the set_param! function does not work correctly after building the model, no idea of where this comes from. Will wait for version 0.19 !
